src/mip/coefficient_strengthening.jl

```diff
@@ -0,0 +1,175 @@
+"""
+Perform coefficient strengthening on each row and on each integer variable.
+
+In particular, for the i-th constraints of the form
+aᵢ x ⩽ bᵢ and let xⱼ be a integer/binary variable
+If aᵢⱼ ⩾ d = bᵢ - Mᵢⱼ - aᵢⱼ (uⱼ - 1) > 0
+where Mᵢⱼ is maximal activity of the i-th row without considering xⱼ
+and uⱼ is upper bound of xⱼ, then we transform
+aᵢⱼ <- aᵢⱼ - d
+bᵢ <- bᵢ - duⱼ .
+
+A similar update rule is applied for the constraints of the form cᵢ ⩽ aᵢ x.
+
+In case the constraint is ranged, i.e, cᵢ ⩽ aᵢ x ⩽ bᵢ, no coefficient strengthening is performed.
+
+The new coefficients are stored by updating ps.pb0 .
+If some coefficient is reduced to 0, it is kept as 0 in both ps.pb0.arows and ps.pb0.acols,
+and the number of nonzeros in each row/column is updated.
+
+If a row/column is reduced to 0 after this step, it will be set to be inactive.
+"""
+
+function maximal_activity(row::Row{T}, ucol::Vector{T}, lcol::Vector{T})::T where {T}
+    sup = zero(T)
+
+    for (j, aij) in zip(row.nzind, row.nzval)
+        if aij > zero(T)
+            sup += aij*ucol[j]
+        elseif aij < zero(T)
+            sup += aij*lcol[j]
+        end
+    end
+    return T(sup)
+end
+
+function minimal_activity(row::Row{T}, ucol::Vector{T}, lcol::Vector{T})::T where {T}
+    inf = zero(T)
+
+    for (j, aij) in zip(row.nzind, row.nzval)
```

Comment:
You need to skip columns that have been removed from the problem. Note that this requires you to pass either `colflag` or `ps` as an argument
```suggestion
    for (j, aij) in zip(row.nzind, row.nzval)
        ps.colflag[j] || continue
```